### PR TITLE
Inhee park ads post 

### PR DESCRIPTION
**Pull request by InheePark**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
     "@angular/compiler-cli": "^14.1.0",
     "@angular/localize": "^14.1.0",
     "@types/jasmine": "~4.0.0",
+    "@types/uuid": "^8.3.4",
```
Comment (EdmundLT) [inline package.json]: uninstall the uuid

File: src/app/ads/post/post.component.html [hunk 2]
```diff
@@ -1 +1,102 @@
...
+          <div
+            class="row d-flex justify-content-center align-items-center h-100"
+          >
+            <div class="col-12 col-md-9 col-md-7 col-xl-6">
+              <div class="card" style="border-radius: 15px">
+                <div class="card-body p-5">
+                  <h2 class="text-uppercase text-center mb-5">
+                    Hello, Post Your Used good!
+                  </h2>
+                  <!-- <% if (!ad.adsId) { %>
```
Comment (EdmundLT) [inline src/app/ads/post/post.component.html:17]: delete the comment

File: src/app/ads/post/post.component.html [hunk 3]
```diff
@@ -1 +1,102 @@
...
+                    <form action="/ads/create" method="post">
+                  <% } else { %>
+                    <form action="/ads/edit/<%= ad.adsId %>" method="post">
+                   <% } %> -->
```
Comment (EdmundLT) [inline src/app/ads/post/post.component.html]: delete the comment line 17 - line 21

File: src/app/ads/post/post.component.ts [hunk 4]
```diff
@@ -1,15 +1,59 @@
 import { Component, OnInit } from '@angular/core';
+import { AdsService } from 'src/app/services/ads.service';
+import Ad from 'src/app/Models/ads.model';
+import User from 'src/app/Models/user.model';
+import { ActivatedRoute } from '@angular/router';
+import { UntypedFormBuilder } from '@angular/forms';
```
Comment (EdmundLT) [inline src/app/ads/post/post.component.ts]: what is this? remove it if it's unused.

File: src/app/ads/post/post.component.ts [hunk 5]
```diff
@@ -1,15 +1,59 @@
...
+    }
+  }
+
+  post(): void{
+    console.log("hello")
+    console.log(this.posting)
+    if(this.posting){
+      this.postInfo.username = "test1";
+      this.adsService.createAd(this.postInfo).subscribe((result) => console.log({result}))
+      this.router.navigate(['/'])
```
Comment (EdmundLT) [inline src/app/ads/post/post.component.ts]: redirect user to the display page

File: src/app/ads/post/post.component.ts [hunk 6]
```diff
@@ -1,15 +1,59 @@
...
+    }else{
+      this.adsService.editAd(this.postInfo.adsId, this.postInfo).subscribe((result) => console.log({result}))
```
Comment (EdmundLT) [inline src/app/ads/post/post.component.ts:53]: how about this? after user submit the edit, where should the user redirect to?

**Review by EdmundLT — COMMENTED** [review]
check the comment, thanks

File: src/app/ads/post/post.component.ts (hunk 4 above)
Comment (InheePark) [reply src/app/ads/post/post.component.ts]: the Ad and ActivatedRoute modules are used
Ad is used in postInfo, 
ActivatedRoute is used in constructor

File: src/app/ads/post/post.component.ts (hunk 6 above)
Comment (InheePark) [reply src/app/ads/post/post.component.ts:53]: I'll also redirect to display page

File: src/app/ads/post/post.component.ts (hunk 5 above)
Comment (InheePark) [reply src/app/ads/post/post.component.ts]: ok

File: src/app/ads/post/post.component.ts (hunk 4 above)
Comment (EdmundLT) [reply src/app/ads/post/post.component.ts]: how about the UntypedFormBuilder

Comment (InheePark) [reply src/app/ads/post/post.component.ts]: I have deleted UntypedFormBuilder

**Review by edmundTang21 — APPROVED** [review]

**Pull request merged**